boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,935 @@
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_ReadFromEfuse(
+    uint16_t *    tx_data,
+    uint16_t *    rx_data,
+    GPIO_TypeDef *chip_select_port,
+    uint16_t      chip_select_pin);
+
+// The structure for each efuse which contains the watchdog-in bit's current
+// value
+static struct EfuseIo aux1_aux2_efuse;
+
+// The SPI handle for the SPI device the E-Fuses are connected to
+static SPI_HandleTypeDef *efuse_spi_handle;
+
+void Io_Efuse_Init(SPI_HandleTypeDef *const hspi)
+{
+    assert(hspi != NULL);
+
+    efuse_spi_handle = hspi;
+
+    aux1_aux2_efuse.wdin_bit_to_set = true;
+}
+
+void Io_Efuse_Aux1Enable(void)
+{
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_Aux1Disable(void)
+{
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_RESET);
+}
+
+void Io_Efuse_Aux2Enable(void)
+{
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_Aux2Disable(void)
+{
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_RESET);
+}
+
+ExitCode Io_Efuse_GetAux1_Aux2Status(enum Efuse_Status *status)
+{
+    return Io_Efuse_Aux1Aux2ReadRegister(SO_STATR_ADDR, (uint16_t *)status);
+}
+
+ExitCode Io_Efuse_GetAux1Faults(enum Efuse_Fault *fault)
+{
+    return Io_Efuse_Aux1Aux2ReadRegister(SO_FAULTR_0_ADDR, (uint16_t *)fault);
+}
+
+ExitCode Io_Efuse_GetAux2Faults(enum Efuse_Fault *fault)
+{
+    return Io_Efuse_Aux1Aux2ReadRegister(SO_FAULTR_1_ADDR, (uint16_t *)fault);
+}
+
+bool Io_Efuse_Aux1_Aux2IsInFaultMode(void)
+{
+    return HAL_GPIO_ReadPin(FSB_AUX1_AUX2_GPIO_Port, FSB_AUX1_AUX2_Pin);
+}
+
+bool Io_Efuse_Aux1_Aux2IsInFailSafeMode(void)
+{
+    return HAL_GPIO_ReadPin(FSOB_AUX1_AUX2_GPIO_Port, FSOB_AUX1_AUX2_Pin);
+}
+
+void Io_Efuse_Aux1Aux2DelatchFaults(void)
+{
+    // Delatch the latchable faults by alternating the inputs high-low-high
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_SET);
+
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_SET);
+}
+
+bool Io_Efuse_GetAux1Current(float *aux1_current)
+{
+    if (Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(
+            AUX1_CURRENT_SENSE_CHANNEL) != EXIT_CODE_OK)
+    {
+        return false;
+    }
+
+    if (HAL_GPIO_ReadPin(
+            CUR_SYNC_AUX1_AUX2_GPIO_Port, CUR_SYNC_AUX1_AUX2_Pin) ==
+        GPIO_PIN_RESET)
+    {
+        *aux1_current = Io_CurrentSense_GetAux1Current();
+        return true;
+    }
+
+    return false;
+}
```

Comment:
I think we should just return a NAN if it times out. Sth like this

```
float Io_Efuse_GetAux1Current(void)
{
    if (Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(
            AUX1_CURRENT_SENSE_CHANNEL) != EXIT_CODE_OK)
    {
        return NAN;
    }

    if (HAL_GPIO_ReadPin(
            CUR_SYNC_AUX1_AUX2_GPIO_Port, CUR_SYNC_AUX1_AUX2_Pin) ==
        GPIO_PIN_RESET)
    {
        return Io_CurrentSense_GetAux1Current();
    }
}
```

Reply:
DONE